File: 5. Longest Palindromic Substring.java
class Solution {
    public String longestPalindrome(String s) {
        String ans = "";
        int n = s.length();

        for(int i = 0; i < n;){
            int l = i, r = i;

            while(r < n - 1 && s.charAt(r) == s.charAt(r + 1)) ++r;
            i = r + 1;

            while(r < n - 1 && l > 0 && s.charAt(l - 1) == s.charAt(r + 1)){
                ++r;
                --l;
            }

            if(ans.length() < r - l + 1){
                ans = s.substring(l, r + 1);
            }
        }


        return ans;
    }

}

File: 5. Longest Palindromic Substring.cpp
#include"LeetCodeLibs.h"

class Solution {
public:
    string longestPalindrome(string s) {
        string res = "";

        for(int i = 0; i < s.size();){
            int l = i;
            int r = i;
            while(r < s.size() - 1 && s[r] == s[r + 1]) ++r;
            i = r + 1;

            while(l > 0 && r < s.size() - 1 && s[l - 1] == s[r + 1]){
                --l;
                ++r;
            }

            if(r - l + 1 > res.size()){
                res = s.substr(l, r - l + 1);
            }
        }

        return res;
    }
};